tree_planter.rb

```diff
@@ -244,8 +245,17 @@ def deploy_tree(endpoint, tree_name, branch_name, repo_url, repo_path, config_ob
 
         Open3.popen2e(deploy_command) do |_stdin, stdout_err, wait_thr|
           # rubocop:disable Lint/AssignmentInCondition
+          output = "There was an error with the tree-planter deployment.\n\n"
+          output << "endpoint:  #{endpoint}\n"
+          output << "tree:      #{tree_name}\n"
+          output << "branch:    #{branch_name}\n"
+          output << "repo_url:  #{repo_url}\n"
+          output << "repo_path: #{repo_path}\n"
+          output << "base:      #{base}\n\n"
+          output << "Attempted to #{deploy_command}\n\n"
```

Comment:
A couple of things:

1. `output` is not really a good name for this as it is actually only the body of a potential email. Maybe adjust the name to `email_body`
2. By removing the first line and rewording the last line this no longer implies that an error has already occured when it has not. It also opens the door for other email use cases later.

```suggestion
          email_body << "endpoint:  #{endpoint}\n"
          email_body << "tree:      #{tree_name}\n"
          email_body << "branch:    #{branch_name}\n"
          email_body << "repo_url:  #{repo_url}\n"
          email_body << "repo_path: #{repo_path}\n"
          email_body << "base:      #{base}\n\n"
          email_body << "Deploy command: #{deploy_command}\n\n"
```

Lastly, what does this set of data give you that you don't get from the `body_content` variable that is already being used / built?

Reply:
Right, `email_body` makes a lot more sense.

I tried to use `body_content` first, but I couldn't seem to find a method to convert it to a string, it always outputs as an object something like `#<Sinatra::Helpers::Stream:0x000055f51495bbb8>` or `#<IO:0x0000559e52bb8c58>`. I'll give it another shot though in case I missed something simple.

Reply:
Ah yes, it's a stream. I forgot.